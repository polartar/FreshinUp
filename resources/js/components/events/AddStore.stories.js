import { storiesOf } from '@storybook/vue'

import AddStore from './AddStore'

const stores = [
  {
    uuid: 'a7936425-485a-4419-9acd-13cdccaed346',
    name: 'Burger Babes',
    type: {
      id: 1,
      name: 'Mobil'
    },
    location: {
      id: 1,
      uuid: 'c4fd0928-b7eb-43ee-871e-e63bfbd0ae7a',
      name: 'Port Gerald'
    },
    store_tags: [
      {
        uuid: '1',
        name: 'minus'
      },
      {
        uuid: '2',
        name: 'hic'
      }
    ],
    assigned: false
  },
  {
    uuid: 'c48fb5d3-37e0-4cb5-bb44-d2d1b5fd97d5',
    name: 'Holy Canoli',
    type: {
      id: 1,
      name: 'Mobil'
    },
    location: {
      id: 2,
      uuid: '00aabd70-aa77-42de-87fd-96449bbd5439',
      name: 'Carminestad'
    },
    store_tags: [
      {
        uuid: '1',
        name: 'minus'
      },
      {
        uuid: '2',
        name: 'sit'
      },
      {
        uuid: '3',
        name: 'hicsdfsdf'
      }
    ],
    assigned: true
  },
  {
    uuid: '790aba97-1eb6-4630-82d9-7bd561256c67',
    name: 'Da Lobsta',
    type: {
      id: 1,
      name: 'Mobil'
    },
    location: {
      id: 2,
      uuid: 'dfb4bd02-c298-4dd4-ab49-2c3f156f2894',
      name: 'New Charlenebury'
    },
    store_tags: [
      {
        uuid: '1',
        name: 'minus'
      },
      {
        uuid: '2',
        name: 'hic'
      },
      {
        uuid: '3',
        name: 'acsfdd'
      },
      {
        uuid: '4',
        name: 'fsdf'
      }
    ],
    assigned: false
  }
]

storiesOf('FoodFleet|event/AddStore', module)
  .add('empty', () => ({
    components: { AddStore },
    template: `
      <v-container>
        <add-store />
      </v-container>
    `
  }))
  .add('with data', () => ({
    components: { AddStore },
    data () {
      return {
        stores: stores
      }
    },
    template: `
      <v-container>
        <add-store :members="stores"/>
      </v-container>
    `
  }))
