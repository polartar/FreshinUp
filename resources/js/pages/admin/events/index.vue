<template>
  <div>
    <v-flex
      d-flex
      align-center
      justify-space-between
      ma-3
    >
      <v-layout
        flex
        align-center
      >
        <h2 class="white--text mr-3">
          {{ pageTitle }}
        </h2>
        <v-btn
          slot="activator"
          color="white"
          @click="eventNew"
        >
          <span class="primary--text">Add New Event</span>
        </v-btn>
      </v-layout>
      <v-flex
        text-xs-right
        sm2
        xs12
      >
        <v-select
          v-model="view"
          :items="views"
          single-line
          solo
          flat
          hide-details
        />
      </v-flex>
    </v-flex>
    <v-divider />
    <template
      v-if="view === 1"
    >
      <filter-sorter
        v-if="!isLoading"
        :statuses="statuses"
        :types="eventTypes"
        @runFilter="filterEvents"
      />
      <event-list
        v-if="!isLoading"
        :events="events"
        :statuses="statuses"
        :is-loading="isLoading || isLoadingList"
        :rows-per-page="pagination.rowsPerPage"
        :page="pagination.page"
        :total-items="pagination.totalItems"
        :sort-by="sorting.sortBy"
        :descending="sorting.descending"
        @paginate="onPaginate"
        @manage-edit="eventEdit"
        @manage-delete="deleteSingle"
        @manage-multiple-delete="multipleDelete"
        @change-status="changeStatusSingle"
        @change-status-multiple="changeStatusMultiple"
      />
    </template>
    <v-layout
      v-else-if="view === 2"
      row
      flex
    >
      <v-flex
        sm3
        xs12
        ml-3
        mt-3
      >
        <v-card>
          <filter-sorter-for-calendar
            v-if="!isLoading"
            :statuses="statuses"
            @runFilter="filterEvents"
          />
        </v-card>
      </v-flex>
      <v-flex
        sm9
        xs12
        ma-3
      >
        <v-card>
          <event-calendar
            v-if="!isLoading"
            :events="events"
            :date="calendarDefaultDate"
            :year-range="calendarYearRange"
            @click-event="eventEdit"
          />
        </v-card>
      </v-flex>
    </v-layout>
    <v-dialog
      v-model="deleteDialog"
      max-width="500"
    >
      <simple-confirm
        :class="{ 'deleting': deletablesProcessing }"
        :title="deleteDialogTitle"
        ok-label="Yes"
        cancel-label="No"
        @ok="onSubmitDelete"
        @cancel="onCancelDelete"
      >
        <div class="py-5 px-2">
          <template v-if="deletablesProcessing">
            <div class="text-xs-center">
              <p class="subheading">
                Processing, please wait...
              </p>
              <v-progress-circular
                :rotate="-90"
                :size="200"
                :width="15"
                :value="deletablesProgress"
                color="primary"
              >
                {{ deletablesStatus }}
              </v-progress-circular>
            </div>
          </template>
          <template v-else>
            <p class="subheading">
              <span v-if="deletables.length < 2">Event</span>
              <span v-else> Events</span>
              : {{ deleteTemp | formatDeleteTitles }}
            </p>
          </template>
        </div>
      </simple-confirm>
    </v-dialog>
  </div>
</template>
<script>
import get from 'lodash/get'
import moment from 'moment'
import { mapActions, mapGetters, mapState } from 'vuex'
import { deletables } from 'fresh-bus/components/mixins/Deletables'
import FilterSorter from '~/components/events/FilterSorter.vue'
import FilterSorterForCalendar from '~/components/events/FilterSorterForCalendar.vue'
import EventList from '~/components/events/EventList.vue'
import EventCalendar from '~/components/events/EventCalendar.vue'
import SimpleConfirm from 'fresh-bus/components/SimpleConfirm.vue'

const INCLUDE = [
  'status',
  'host',
  'location',
  'manager',
  'type',
  'venue'
]

export default {
  layout: 'admin',
  components: {
    EventList,
    EventCalendar,
    FilterSorter,
    FilterSorterForCalendar,
    SimpleConfirm
  },
  filters: {
    formatDeleteTitles (value) {
      return value.map(item => item.name).join(', ')
    }
  },
  mixins: [deletables],
  data () {
    const currentMoment = moment()
    return {
      pageTitle: 'Events',
      deleteDialog: false,
      deleteTemp: [],
      deletablesProcessing: false,
      deletablesProgress: 0,
      deletablesStatus: '',
      lastFilterParams: {},
      calendarDefaultDate: currentMoment.format('YYYY-MM-DD'),
      calendarYearRange: [currentMoment.year() - 2, currentMoment.year() + 2],
      view: 1,
      views: [
        { value: 1, text: 'List view' },
        { value: 2, text: 'Calendar view' }
      ]
    }
  },
  computed: {
    isLoadingList () {
      return get(this.$store, 'state.events.pending.items', true)
    },
    ...mapGetters(['currentUser']),
    ...mapGetters('events', {
      events: 'items',
      pagination: 'pagination',
      sorting: 'sorting',
      sortBy: 'sortBy'
    }),
    ...mapGetters('eventStatuses', { 'statuses': 'items' }),
    ...mapGetters('eventTypes', { 'eventTypes': 'items' }),
    ...mapGetters('page', ['isLoading']),
    ...mapState('events', ['sortables']),
    deleteDialogTitle () {
      return this.deleteTemp.length < 2 ? 'Are you sure you want to delete this event?' : 'Are you sure you want to delete the following events?'
    },
    role () {
      if (this.currentUser.isAdmin) return 'admin'
      if (this.currentUser.type === 1) return 'supplier'
      if (this.currentUser.type === 2) return 'host'
      return 'admin'
    }
  },
  methods: {
    ...mapActions('page', {
      setPageLoading: 'setLoading'
    }),
    eventNew () {
      this.$router.push({ path: '/admin/events/new' })
    },
    eventEdit (event) {
      this.$router.push({ path: '/admin/events/' + event.uuid + '/edit' })
    },
    deleteSingle (event) {
      this.deleteTemp = [event]
      this.deleteDialog = true
    },
    multipleDelete (events) {
      this.deleteTemp = events
      this.deleteDialog = true
    },
    changeStatusSingle (statusId, event) {
      this.$store.dispatch('events/patchItem', {
        data: { status_id: statusId },
        params: { id: event.uuid }
      }).then(() => {
        this.filterEvents(this.lastFilterParams)
      })
    },
    changeStatusMultiple (statusId, events) {
      events.forEach((event) => {
        this.changeStatusSingle(statusId, event)
      })
    },
    async onSubmitDelete () {
      this.deletablesProcessing = true
      this.deletablesProgress = 0
      this.deletablesStatus = ''
      let dispatcheables = []

      this.deleteTemp.forEach((event) => {
        dispatcheables.push(this.$store.dispatch('events/deleteItem', {
          getItems: false,
          params: { id: event.uuid }
        }))
      })

      let chunks = this.chunk(dispatcheables, this.deleteTempParrallelRequest)
      let doneCount = 0

      for (let i in chunks) {
        await Promise.all(chunks[i])
        doneCount += chunks[i].length
        this.deleteTempStatus = doneCount + ' / ' + this.deleteTemp.length + ' Done'
        this.deleteTempProgress = doneCount / this.deleteTemp.length * 100
        await this.sleep(this.deletablesSleepTime)
      }

      this.filterEvents(this.lastFilterParams)
      await this.sleep(500)
      this.deletablesProcessing = false
      this.deleteDialog = false
    },
    onCancelDelete () {
      this.deleteDialog = false
      this.deleteTemp = []
    },
    onPaginate (value) {
      this.$store.dispatch('events/setPagination', value)
      this.$store.dispatch('events/getItems', { params: { include: INCLUDE } })
    },
    filterEvents (params) {
      this.lastFilterParams = params
      this.$store.dispatch('events/setFilters', {
        ...this.$route.query,
        ...this.lastFilterParams
      })
      this.$store.dispatch('events/getItems', { params: { include: INCLUDE } })
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('events/setFilters', {
      ...vm.$route.query,
      ...this.lastFilterParams
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('eventStatuses/getItems'),
      vm.$store.dispatch('eventTypes/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => {
        vm.setPageLoading(false)
      })
  }
}
</script>
