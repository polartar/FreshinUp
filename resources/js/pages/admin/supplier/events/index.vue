<script>
import Page, { INCLUDE } from '~/pages/admin/events/index.vue'
import { mapGetters } from 'vuex'
export default {
  extends: Page,
  computed: {
    ...mapGetters('suppliers/events', {
      events: 'items',
      pagination: 'pagination',
      sorting: 'sorting',
      sortBy: 'sortBy',
      sortables: 'sortables'
    })
  },
  methods: {
    eventNew () {
      this.$router.push({ path: '/admin/supplier/events/new' })
    },
    eventEdit (event) {
      this.$router.push({ path: '/admin/supplier/events/' + event.uuid + '/edit' })
    },
    onPaginate (value) {
      this.$store.dispatch('suppliers/events/setPagination', value)
      this.$store.dispatch('suppliers/events/getItems', {
        params: {
          include: INCLUDE,
          supplierId: this.currentUser.uuid
        }
      })
    },
    filterEvents (params) {
      this.lastFilterParams = params
      this.$store.dispatch('suppliers/events/setFilters', {
        ...this.$route.query,
        ...this.lastFilterParams
      })
      this.$store.dispatch('suppliers/events/getItems', {
        params: {
          include: INCLUDE,
          supplierId: this.currentUser.uuid
        }
      })
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('suppliers/events/setFilters', {
      ...vm.$route.query,
      ...this.lastFilterParams
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('eventStatuses/getItems'),
      vm.$store.dispatch('eventTypes/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => {
        vm.setPageLoading(false)
      })
  }
}
</script>
