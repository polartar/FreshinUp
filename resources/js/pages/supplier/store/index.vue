
<script>
import stores from '~/pages/admin/fleet-members/index.vue'
import { mapGetters } from 'vuex'
const INCLUDE = [
  'tags',
  'addresses',
  'status',
  'owner',
  'type'
]

export default {
  extends: stores,
  computed: {
    ...mapGetters(['currentUser'])
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('stores/setFilters', {
        ...vm.$route.query,
        ...this.lastFilterParams,
        owner_uuid: vm.currentUser.uuid
      }),
      vm.$store.dispatch('storeStatuses/getItems', {
        params: { include: INCLUDE }
      })
    ])
      .then(() => {
        if (next) next()
      })
      .catch((error) => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
