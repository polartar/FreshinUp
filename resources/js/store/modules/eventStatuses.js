import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default ({ items, item }) => {
  return makeRestStore('foodfleet/event-statuses', { items, item })
}
