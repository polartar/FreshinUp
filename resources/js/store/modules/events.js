import makeRestStore from 'fresh-bus/store/utils/makeRestStore'
import omit from 'lodash/omit'

export default ({ items, item }) => {
  const store = makeRestStore(
    'events',
    { items, item },
    {
      itemsPath: () => `/foodfleet/events`,
      itemPath: ({ id }) => `/foodfleet/events/${id}`
    }
  )
  const __createItem = store.actions.createItem
  store.actions = {
    ...store.actions,
    createItem (context, payload) {
      payload.data = omit(payload.data, ['event_recurring_checked', 'event_tags', 'host', 'venue', 'manager', 'status'])
      return __createItem(context, payload)
    }
  }

  return {
    namespaced: true,
    ...store,
    modules: {
      stores: makeRestStore(
        'stores',
        { items, item },
        {
          itemsPath: ({ eventId }) => `/foodfleet/events/${eventId}/stores`,
          itemPath: ({ eventId, id }) => `/foodfleet/events/${eventId}/stores/${id}`
        }
      )
    }
  }
}
