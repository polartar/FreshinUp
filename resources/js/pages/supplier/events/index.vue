
<script>
import { mapActions, mapGetters } from 'vuex'
import events from '~/pages/admin/events/index.vue'

const INCLUDE = [
  'status',
  'host',
  'location',
  'manager',
  'type',
  'venue'
]

export default {
  extends: events,
  computed: {
    ...mapGetters(['currentUser'])
  },
  methods: {
    ...mapActions('page', {
      setPageLoading: 'setLoading'
    }),
    onPaginate (value) {
      this.$store.dispatch('events/setPagination', value)
      this.$store.dispatch('events/getItems', { params: { include: INCLUDE, manager_uuid: this.currentUser.uuid } })
    },
    filterEvents (params) {
      this.lastFilterParams = params
      this.$store.dispatch('events/setFilters', {
        ...this.$route.query,
        ...this.lastFilterParams,
        manager_uuid: this.currentUser.uuid
      })
      this.$store.dispatch('events/getItems', { params: { include: INCLUDE, manager_uuid: this.currentUser.uuid } })
    }
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.$store.dispatch('events/setFilters', {
      ...vm.$route.query,
      ...this.lastFilterParams,
      manager_uuid: vm.currentUser.uuid
    })
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('eventStatuses/getItems'),
      vm.$store.dispatch('eventTypes/getItems')
    ])
      .then(() => {
        if (next) next()
      })
      .catch(error => console.error(error))
      .then(() => {
        vm.setPageLoading(false)
      })
  }
}
</script>
