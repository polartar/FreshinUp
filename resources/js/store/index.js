import { createStore } from 'fresh-bus/store'

import page from './modules/page'
import devices from './modules/devices'
import financialModifiers from './modules/financialModifiers'
import financialReports from './modules/financialReports'
import paymentTypes from './modules/paymentTypes'
import financialsummary from './modules/financialsummary'
import squares from './modules/squares'
import documents from './modules/documents'
import documentStatuses from './modules/documentStatuses'
import documentTypes from './modules/documentTypes'
import transactions from './modules/transactions'
import companyDetails from './modules/companyDetails'
import eventSummary from './modules/eventSummary'
import eventTypes from './modules/eventTypes'
import eventHistories from './modules/eventHistories'
import venues from './modules/venues'
import stores from './modules/stores'
import storeStatuses from './modules/storeStatuses'
import storeTypes from './modules/storeTypes'
import companies from './modules/companies'

export default (initialState = {}) => {
  return createStore(
    initialState,
    {
      modules: {
        page: page({}),
        devices: devices({}),
        financialModifiers: financialModifiers({}),
        financialReports: financialReports({}),
        paymentTypes: paymentTypes({}),
        financialsummary: financialsummary({}),
        squares: squares({}),
        documents: documents({}),
        documentStatuses: documentStatuses({}),
        documentTypes: documentTypes({}),
        transactions: transactions({}),
        companyDetails: companyDetails({}),
        eventSummary: eventSummary({}),
        eventTypes: eventTypes({}),
        eventHistories: eventHistories({}),
        venues: venues({}),
        stores: stores({}),
        storeStatuses: storeStatuses({}),
        storeTypes: storeTypes({}),
        companies: companies({})
      }
    }
  )
}
