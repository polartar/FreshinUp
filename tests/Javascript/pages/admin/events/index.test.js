import { shallowMount } from '@vue/test-utils'
import { createStore } from 'fresh-bus/store'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import { FIXTURE_EVENTS } from 'tests/__data__/events'
import Component from '~/pages/admin/events/index.vue'
import events from '~/store/modules/events'
import eventStatuses from '~/store/modules/eventStatuses'

describe('Admin Events Page', () => {
  let localVue, mock, store, actions
  describe('Mount', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
    })

    afterEach(() => {
      mock.restore()
    })

    test('snapshot', async () => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
        .onGet('api/foodfleet/events').reply(200, FIXTURE_EVENTS)
        .onAny()
        .reply(config => {
          console.warn('No mock match for ' + config.url, config)
          return [404, {}]
        })
      const store = createStore({}, {
        modules: {
          events: events({}),
          eventStatuses: eventStatuses({})
        }
      })
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })
      // Action: change State Machine's state
      await wrapper.vm.$store.dispatch('page/setLoading', false)
      await wrapper.vm.$nextTick()
      await wrapper.vm.$store.dispatch('events/getItems')
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      const eventModule = events({})
      localVue = vue.localVue
      actions = {
        patchItem: jest.fn(),
        deleteItem: jest.fn(),
        getItems: jest.fn(),
        setFilters: jest.fn()
      }
      store = createStore({
        events: {
          items: FIXTURE_EVENTS
        }
      }, {
        modules: {
          events: { ...eventModule, actions: { ...eventModule.actions, ...actions } }
        }
      })
    })

    afterEach(() => {
      mock.restore()
    })

    test('changeStatusSingle function change event status', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.changeStatusSingle(2, { uuid: 'mock uuid' })
      const data = { data: { status_id: 2 }, params: { id: 'mock uuid' } }
      expect(actions.patchItem).toHaveBeenCalled()
      expect(actions.patchItem.mock.calls).toHaveLength(1)
      expect(actions.patchItem.mock.calls[0][1]).toEqual(data)
    })

    test('changeStatusMultiple function change event status for each', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.changeStatusMultiple(3, [{ uuid: 'mock uuid 1' }, { uuid: 'mock uuid 2' }])

      expect(actions.patchItem).toHaveBeenCalled()
      expect(actions.patchItem.mock.calls).toHaveLength(2)

      const firstData = { data: { status_id: 3 }, params: { id: 'mock uuid 1' } }
      const secondData = { data: { status_id: 3 }, params: { id: 'mock uuid 2' } }

      expect(actions.patchItem.mock.calls[0][1]).toEqual(firstData)
      expect(actions.patchItem.mock.calls[1][1]).toEqual(secondData)
    })

    test('onSubmitDelete function delete events in deleteTemp ', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.setData({ deleteTemp: [{ uuid: 'mock uuid 1' }, { uuid: 'mock uuid 2' }] })
      wrapper.vm.onSubmitDelete()

      expect(actions.deleteItem).toHaveBeenCalled()
      expect(actions.deleteItem.mock.calls).toHaveLength(2)

      const firstData = { getItems: false, params: { id: 'mock uuid 1' } }
      const secondData = { getItems: false, params: { id: 'mock uuid 2' } }

      expect(actions.deleteItem.mock.calls[0][1]).toEqual(firstData)
      expect(actions.deleteItem.mock.calls[1][1]).toEqual(secondData)
    })

    test('filterEvents function filter events with filterParams ', async () => {
      const $route = {
        query: {}
      }
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store,
        mocks: {
          $route
        }
      })

      const params = { name: 'mock filter name' }
      wrapper.vm.filterEvents(params)
      expect(actions.setFilters).toHaveBeenCalled()
      expect(actions.setFilters.mock.calls).toHaveLength(1)
      expect(actions.setFilters.mock.calls[0][1]).toEqual(params)

      expect(actions.getItems).toHaveBeenCalled()
      expect(actions.getItems.mock.calls).toHaveLength(1)
    })

    test('onPaginate function change paginate', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.onPaginate({
        rowsPerPage: 2,
        totalItems: 5,
        page: 2
      })
      expect(wrapper.vm.pagination.rowsPerPage).toBe(2)
    })

    test('deleteSingle function change deleteTemp', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.deleteSingle({ id: 1 })
      expect(wrapper.vm.deleteTemp[0].id).toBe(1)
      expect(wrapper.vm.deleteDialog).toBeTruthy()
    })

    test('multipleDelete function change deleteTemp', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.multipleDelete([{ id: 1 }])
      expect(wrapper.vm.deleteTemp[0].id).toBe(1)
      expect(wrapper.vm.deleteDialog).toBeTruthy()
    })

    test('onCancelDelete function change deleteTemp', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.onCancelDelete()
      expect(wrapper.vm.deleteTemp).toHaveLength(0)
    })
  })
})
