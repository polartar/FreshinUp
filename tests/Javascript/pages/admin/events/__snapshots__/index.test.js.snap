// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Events Page Mount snapshot 1`] = `
<div>
  <v-flex-stub
    align-center=""
    d-flex=""
    justify-space-between=""
    ma-3=""
    tag="div"
  >
    <v-layout-stub
      align-center=""
      flex=""
      tag="div"
    >
      <h2
        class="white--text mr-3"
      >
        
        Events
      
      </h2>
       
      <v-btn-stub
        activeclass="v-btn--active"
        color="white"
        slot="activator"
        tag="button"
        type="button"
      >
        <span
          class="primary--text"
        >
          Add New Event
        </span>
      </v-btn-stub>
    </v-layout-stub>
     
    <v-flex-stub
      sm2=""
      tag="div"
      text-xs-right=""
      xs12=""
    >
      <v-select-stub
        flat=""
        hide-details=""
        items="[object Object],[object Object]"
        single-line=""
        solo=""
        value="1"
      />
    </v-flex-stub>
  </v-flex-stub>
   
  <v-divider-stub />
   
  <!---->
   
  <!---->
   
  <v-dialog-stub
    contentclass=""
    maxwidth="500"
    origin="center center"
    transition="dialog-transition"
    width="auto"
  >
    <simple-confirm-stub
      cancellabel="No"
      class=""
      oklabel="Yes"
      title="Are you sure you want to delete this event?"
    >
      <div
        class="py-5 px-2"
      >
        <p
          class="subheading"
        >
          <span>
            Event
          </span>
          
            : 
          
        </p>
      </div>
    </simple-confirm-stub>
  </v-dialog-stub>
</div>
`;
